src/main/java/org/quantumbadger/redreader/common/PrefsUtility.java

```diff
@@ -865,14 +867,52 @@ public static boolean network_tor(final Context context, final SharedPreferences
+			this.itemPref = itemPref;
+			this.stringRes = stringRes;
+			this.defaultValue = defaultValue;
+		}
+	}
 
-		final EnumSet<OptionsMenuUtility.OptionsMenuItemsPref> result = EnumSet.noneOf(OptionsMenuUtility.OptionsMenuItemsPref.class);
-		for(String s : strings) result.add(OptionsMenuUtility.OptionsMenuItemsPref.valueOf(General.asciiUppercase(s)));
+	public static Map<Enum<OptionsMenuUtility.AppbarItemsPref>, Integer> pref_menus_appbar_items(final Context context, final SharedPreferences sharedPreferences) {
 
-		return result;
+		final AppbarItemInfo[] appbarItemsInfo = new AppbarItemInfo[] {
+			new AppbarItemInfo(OptionsMenuUtility.AppbarItemsPref.SORT, R.string.pref_menus_appbar_sort_key, MenuItem.SHOW_AS_ACTION_ALWAYS),
+			new AppbarItemInfo(OptionsMenuUtility.AppbarItemsPref.REFRESH, R.string.pref_menus_appbar_refresh_key, MenuItem.SHOW_AS_ACTION_ALWAYS),
+			new AppbarItemInfo(OptionsMenuUtility.AppbarItemsPref.PAST, R.string.pref_menus_appbar_past_key, MenuItem.SHOW_AS_ACTION_NEVER),
+			new AppbarItemInfo(OptionsMenuUtility.AppbarItemsPref.SUBMIT_POST, R.string.pref_menus_appbar_submit_post_key, MenuItem.SHOW_AS_ACTION_NEVER),
+			new AppbarItemInfo(OptionsMenuUtility.AppbarItemsPref.PIN, R.string.pref_menus_appbar_pin_key, MenuItem.SHOW_AS_ACTION_NEVER),
+			new AppbarItemInfo(OptionsMenuUtility.AppbarItemsPref.BLOCK, R.string.pref_menus_appbar_block_key, MenuItem.SHOW_AS_ACTION_NEVER),
+			new AppbarItemInfo(OptionsMenuUtility.AppbarItemsPref.SUBSCRIBE, R.string.pref_menus_appbar_subscribe_key, MenuItem.SHOW_AS_ACTION_NEVER),
+			new AppbarItemInfo(OptionsMenuUtility.AppbarItemsPref.SIDEBAR, R.string.pref_menus_appbar_sidebar_key, MenuItem.SHOW_AS_ACTION_NEVER),
+			new AppbarItemInfo(OptionsMenuUtility.AppbarItemsPref.ACCOUNTS, R.string.pref_menus_appbar_accounts_key, MenuItem.SHOW_AS_ACTION_NEVER),
+			new AppbarItemInfo(OptionsMenuUtility.AppbarItemsPref.THEME, R.string.pref_menus_appbar_theme_key, MenuItem.SHOW_AS_ACTION_NEVER),
+			new AppbarItemInfo(OptionsMenuUtility.AppbarItemsPref.SETTINGS, R.string.pref_menus_appbar_settings_key, MenuItem.SHOW_AS_ACTION_NEVER),
+			new AppbarItemInfo(OptionsMenuUtility.AppbarItemsPref.CLOSE_ALL, R.string.pref_menus_appbar_close_all_key, OptionsMenuUtility.DO_NOT_SHOW),
+			new AppbarItemInfo(OptionsMenuUtility.AppbarItemsPref.REPLY, R.string.pref_menus_appbar_reply_key, MenuItem.SHOW_AS_ACTION_NEVER),
+			new AppbarItemInfo(OptionsMenuUtility.AppbarItemsPref.SEARCH, R.string.pref_menus_appbar_search_key, MenuItem.SHOW_AS_ACTION_NEVER)
+		};
+
+
+		Map<Enum<OptionsMenuUtility.AppbarItemsPref>, Integer> appbarItemsPrefs = new HashMap<>();
+
+		for(AppbarItemInfo item : appbarItemsInfo) {
+			try {
+				appbarItemsPrefs.put(item.itemPref, Integer.parseInt(getString(item.stringRes,
+						Integer.toString(item.defaultValue),
+						context,
+						sharedPreferences)));
+			} catch(Throwable e) {
```

Comment:
Catching `Throwable` isn't ideal -- is this just to catch exceptions thrown by `Integer.parseInt`, and also maybe null pointer exceptions? If so, it should be possible to specify those directly:

```java
} catch(final NumberFormatException | NullPointerException e) {
```